### PR TITLE
fix(dataframes): Fix map_series not being able to map values to nan

### PR DESCRIPTION
**Pull request by pabloarosado**
Currently, `dataframes.map_series()` is not able to map values to nan. The reason is as follows:
First, it applies the mapping using `map`, which replaces all values that were not in the mapping by nan.
So, if `make_unmapped_values_nan` is False, we need to bring those nans back to their original values, which is what we currently do.
However, we are not considering that one may actually want to map certain values to nan.
In other words, if I create a mapping {"bad_value": np.nan}, I want `dataframes.map_series` to map "bad_values" to nan.
This PR fixes that issue.

Note: The reason why we created this function is because pandas' `replace` was too slow on big dataframes. We should check that this fix is not making `map_series` too slow.

### CODE REVIEW

**Review by Marigold — APPROVED** [review]
Unit test would be nice, but up to you.

**Comment by pabloarosado** [comment]
> Unit test would be nice, but up to you.

I have just added one test for this new case.
